WordPress/src/main/java/org/wordpress/android/ui/prefs/notifications/NotificationsSettingsDialogFragment.kt

```diff
@@ -0,0 +1,399 @@
+package org.wordpress.android.ui.prefs.notifications
+
+import android.annotation.SuppressLint
+import android.app.Dialog
+import android.content.DialogInterface
+import android.os.Bundle
+import android.text.TextUtils
+import android.view.View
+import android.view.ViewGroup
+import android.widget.CompoundButton
+import android.widget.LinearLayout
+import android.widget.ScrollView
+import android.widget.TextView
+import androidx.appcompat.app.ActionBar
+import androidx.appcompat.app.AlertDialog
+import androidx.appcompat.widget.SwitchCompat
+import androidx.core.content.ContextCompat
+import androidx.fragment.app.DialogFragment
+import com.google.android.material.dialog.MaterialAlertDialogBuilder
+import org.json.JSONException
+import org.json.JSONObject
+import org.wordpress.android.R
+import org.wordpress.android.databinding.NotificationsSettingsSwitchBinding
+import org.wordpress.android.models.NotificationsSettings
+import org.wordpress.android.ui.prefs.AppPrefs
+import org.wordpress.android.util.AppLog
+import org.wordpress.android.util.JSONUtils
+import org.wordpress.android.util.extensions.getDrawableFromAttribute
+
+class NotificationsSettingsDialogFragment(
```

Comment:
I would advocate toward using fragment arguments instead of custom constructor parameters. I found [this article useful on how to do this](https://medium.com/@helmersebastian/why-to-use-fragment-arguments-in-android-fbaa375d08c3). [This example of a `DialogFragment`](https://github.com/wordpress-mobile/WordPress-Android/blob/trunk/WordPress/src/main/java/org/wordpress/android/ui/posts/PostNotificationScheduleTimeDialogFragment.kt#L91) using this pattern inside the project can be useful.